### PR TITLE
Clean up and fix issues with env configuration and .env.template

### PR DESCRIPTION
**Pull request by erik-megarad**
### Background

The configuration available is sometimes not documented, frequently doesn't make sense, has unused settings, amongst other issues.

The `.env.template` file is hard to read and as a new user it's really difficult to figure out what's going on.

### Changes

- Clean up and standardize format of the .env.template file. Commented-out environment variables should *always* be set to the default value, and not an example value.
- Add a separate doc page for documentation of config variables that either did not exist or lived in the .env.template
- Rework configuration for image generation as it was messy and confusing.
- Set audio to text provider configuration to be the same as the other providers
- Rename `CUSTOM_SEARCH_ENGINE_ID` to `GOOGLE_CUSTOM_SEARCH_ENGINE_ID`
- Remove Twitter configuration from the .env template since it is no longer part of Auto-GPT core
- Rework configuration for text-to-speech so that the provider isn't magically divined from other settings

### Documentation

It's all docs

### Test Plan

Unit tests, setting values and running myself

### PR Quality Checklist
- [ ] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [ ] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes. 
- [ ] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports autogpt tests --in-place
    ```

### CODE REVIEW

File: autogpt/config/config.py [hunk 1]
```diff
@@ -59,7 +59,7 @@ def __init__(self) -> None:
             "PROMPT_SETTINGS_FILE", "prompt_settings.yaml"
         )
         self.fast_llm_model = os.getenv("FAST_LLM_MODEL", "gpt-3.5-turbo")
-        self.smart_llm_model = os.getenv("SMART_LLM_MODEL", "gpt-4")
+        self.smart_llm_model = os.getenv("SMART_LLM_MODEL", "gpt-3.5-turbo")
```
Comment (erik-megarad) [inline autogpt/config/config.py:62]: This is a default change, but I think it's required. The vast majority of people don't have gpt-4 and we get complaints about it all the time.

Comment (waynehamadi) [reply autogpt/config/config.py:62]: man you're so right, good catch, I completely forgot about that setting.

**Review by merwanehamadi — APPROVED** [review]

File: .env.template [hunk 2]
```diff
@@ -18,67 +24,41 @@
...
-## autogpt.commands.execute_code
-## autogpt.commands.file_operations
-## autogpt.commands.git_operations
-## autogpt.commands.google_search
-## autogpt.commands.image_gen
-## autogpt.commands.improve_code
-## autogpt.commands.web_selenium
-## autogpt.commands.write_tests
-## autogpt.app
-## autogpt.commands.task_statuses
```
Comment (Pwuts) [inline .env.template:40]: In the user guide ([here](https://docs.agpt.co/usage/#:~:text=You%20can%20find%20the%20list%20of%20categories%20in%20your%20.env.template)):
> You can find the list of categories in your .env.template

Where can users find this list now?

Comment (erik-megarad) [reply .env.template:40]: There is a note in the `options.md` that says to look in the `commands` directory.

I expect to be changing this soon.

**Pull request merged**